### DISCUSSION TITLE
My github isn't working. Please help me.

### DISCUSSION THREAD

**Discussion by bsce22030** [community/community#49815] [New to GitHub] [answered]
My github can't commit to main.
It shows this error. 
Author identity unknown

*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"

to set your account's default identity.
Omit --global to set the identity only in this repository.

fatal: unable to auto-detect email address (got 'gng@DESKTOP-R4H9NU7.(none)')

**Comment by maahaMaya** [comment]
Step 1: set user email
In the terminal run the below command with your email id for git,

git config --global user.email "you@example.com"

Step 2: set user name
In the terminal run the below command with your email id for git,

git config --global user.name "user name"

**Reply by bsce22030** [reply]
ty

**Comment by bbkx226** [comment, marked as answer]
The error message suggests that Git is unable to detect your email address and author identity. To resolve this issue, you need to set your email address and username in Git. Here are the steps you can follow:

Open the command prompt or terminal on your computer.

Run the following commands, replacing "[you@example.com](mailto:you@example.com)" with your email address and "Your Name" with your username:



```
git config --global user.email "you@example.com"
git config --global user.name "Your Name"
```

This will set your account's default identity globally, meaning that it will be used for all repositories on your computer. If you want to set your identity only for the current repository, remove the --global flag from the command.

Try to commit your changes again. This time, Git should recognize your identity and allow you to commit to the main branch.
If you still encounter issues, make sure that you have write access to the repository and that you have pulled any changes made by other contributors before attempting to commit.

**Reply by bsce22030** [reply]
it worked. ty

**Reply by bbkx226** [reply]
you're welcomed

**Reply by yanoyama** [reply]
Thank you very much

**Reply by Zarakani** [reply]
It's worked..tq

**Reply by EAPwavia1** [reply]
Yes I have used the same procedure but when i get to the commit level where I do git add . and git commit -m then the message, I get the feedback that unable to autodetect email address. Please help me now.

**Reply by n0tR3tr0-code** [reply]
> Yes I have used the same procedure but when i get to the commit level where I do git add . and git commit -m then the message, I get the feedback that unable to autodetect email address. Please help me now.

same shit, i dont understand why

**Comment by rayhanalshorif133** [comment]
In local Metchine,
Type here to search> Credential Manager > Windows Credential > Github.com > Edit

**Comment by sutiliza** [comment]
Please! #AccountStatusReviewed

**Comment by DivyaKarade** [comment]
The error occurs because Git doesn't know your name and email. Fix it by running:

git config --global user.email "your-email@example.com"
git config --global user.name "Your Name"

Or set it for the specific repository without --global. This will allow Git to commit with the correct identity.

**Comment by DivyaKarade** [comment]
The error means Git doesn't know your user identity (email and name). To fix it, run these commands:

For global setup (all repos):

git config --global user.email "your-email@example.com"
git config --global user.name "Your Name"

For local repo only:

git config user.email "your-email@example.com"
git config user.name "Your Name"

This will resolve the issue and allow you to commit to the main branch.
